src/Core/Addon/Theme/ThemeRepository.php

```diff
@@ -52,18 +52,22 @@ public function __construct(ConfigurationInterface $configuration, Filesystem $f
     public function getInstanceByName($name)
     {
         $dir = $this->appConfiguration->get('_PS_ALL_THEMES_DIR_').$name;
+        $themeCachePath = $this->appConfiguration->get(
+                '_PS_CACHE_DIR_'
+            ) . 'themes/' . $name;
 
-        $jsonConf = '';
+        @mkdir($themeCachePath, 0777, true);
         if ($this->shop) {
-            $jsonConf = $this->appConfiguration->get(
-                    '_PS_CACHE_DIR_'
-                ) . 'themes/' . $name . '/shop' . $this->shop->id . '.json';
+            $jsonConf = $themeCachePath . '/shop' . $this->shop->id . '.json';
+        } else {
+            $jsonConf = $themeCachePath . '/theme.json';
         }
 
         if ($this->filesystem->exists($jsonConf)) {
             $data = $this->getConfigFromFile($jsonConf);
         } else {
             $data = $this->getConfigFromFile($dir.'/config/theme.yml');
+            file_put_contents($jsonConf, json_encode($data));
```

Comment:
Can you please use the `dumpFile` function from [Filesystem](https://symfony.com/doc/2.8/components/filesystem.html#dumpfile)?